tests/utils/_a_sequence.py

```diff
@@ -0,0 +1,42 @@
+# tests/timeseries/test_resampling.py:93: License: MIT
+# Copyright © 2022 Frequenz Energy-as-a-Service GmbH
+
+"""Helper class to compare two sequences without caring about the underlying type."""
+
+
+from typing import Any, Sequence
```

Comment:
I found it is deprecated:
https://docs.python.org/3.10/library/typing.html#typing.Sequence

If we would use python >= 3.10, then mypy supports generics types.
https://peps.python.org/pep-0585/#implementation
https://www.scivision.dev/mypy-pep-585-604/

Reply:
About this I was thinking for a while that we probably also need a `_compat` module to deal with these issues between different python versions. Because in 3.8 is not deprecated and I'm not sure if in 3.8 `collections.abc.Sequence` support indenxing (so it can be used as a type hint), I will need to check.

Another thing I would put in the `_compat` module is `anext()` (although in the channels repo I think, I don't think we use it here).

Reply:
Yeah, with Python 3.8 I get this error if I use `collections.abc.Sequence`:
```
tests/utils/_a_sequence.py:29:23: E1136: Value 'Sequence' is unsubscriptable (unsubscriptable-object)
```

Reply:
Could you please send url to the _compat module?
I can't find it in google :)

Reply:
It doesn't exist. It is just an idea I had in mind. After this I actually tried to add one for `Sequence`, but it is not that simple for this case because even for the same Python version (3.8) you might need to import `Sequence` from different places depending of how you want to use it (a type hint or a ABC), so I couldn't decide how to name stuff, as it it lives in `compat`, we only need to import that for type hinting, in which case it would make sense to call it `compat.typing` to make it more clear, but then it feels weird to use `typing` when more modern pythons need less and less the use of the `typing` module.

I know it is nitpicking, but I couldn't decide, and we already are importing `Sequence` from `typing` in other parts of the SDK code so I figure we leave it like it is if we didn't have any issues with it so far.

Reply:
> Yeah, with Python 3.8 I get this error if I use collections.abc.Sequence:

it looks like we can do `from __future__ import annotations` and it will start working in 3.8

Reply:
Oh really? That's cool! I'll try it out!

Reply:
I will try it out but I have very little hope, `annotations` seems to be only about forward references: https://peps.python.org/pep-0563/

Reply:
It is mentioned here: https://peps.python.org/pep-0585/#implementation

Reply:
Then awesome if it means we don't have to import stuff from `typing` for built-in types either :partying_face:

Reply:
It is *almost* awesome. It does work for type hinting, but it doesn't work for other usages, like type aliases `MyType = Sequence[int]` or `cast(Sequence[int], s)` will still fail if `Sequence` doesn't come from `typing`.

But well... Is better than nothing.